lib/ansible/modules/windows/win_user_right.ps1

```diff
@@ -0,0 +1,354 @@
+            $new_users += "*$user_sid"
+        }
+    } else {
+        foreach ($user in $users) {
+            $user_sid = Get-SID -account_name $user
+            $new_users += "*$user_sid"
+        }
+    }
+
+    ,$new_users
+}
+
+Function Build-ExistingUserList($existing_users) {
+    $users = @()
+    foreach ($existing_user in ($existing_users -split ",")) {
+        if ($existing_user.StartsWith("*S")) {
+            $users += $existing_user
+        } else {
+            $user_sid = Get-SID -account_name $existing_user
+            $users += "*$user_sid"
+        }
+    }
+
+    ,$users
+}
+
+Function ConvertTo-Ini($ini) {
+    $content = @()
+    foreach ($key in $ini.GetEnumerator()) {
+        $section = $key.Name
+        $values = $key.Value
+
+        $content += "[$section]"
+        foreach ($value in $values.GetEnumerator()) {
+            $value_key = $value.Name
+            $value_value = $value.Value
+
+            if ($value_value -ne $null) {
+                $content += "$value_key = $value_value"
+            }
+        }
+    }
+
+    $content -join "`r`n"
+}
+
+Function ConvertFrom-Ini($file_path) {
+    $ini = @{}
+    $contents = Get-Content -Path $file_path -Encoding Unicode
+    foreach ($line in $contents) {
+        switch -Regex ($line) {
+            "^\[(.+)\]" {
+                $section = $matches[1]
+                $ini.$section = @{}
+            }
+            "(.+?)\s*=(.*)" {
+                $name = $matches[1].Trim()
+                $value = $matches[2].Trim()
+                $ini.$section.$name = $value
+            }
+        }
+    }
+
+    $ini
+}
+
+### Main code below ###
+
+$ini_right_key = "Privilege Rights"
+$secedit_ini_path = [IO.Path]::GetTempFileName()
+# while this will technically make a change to the system in check mode by
+# creating a new file, we need these values to be able to do anything
+# substantial in check mode
+$export_result = Run-SecEdit -arguments @("/export", "/cfg", $secedit_ini_path, "/quiet")
+
+# check the return code and if the file has been populated, otherwise error out
+if (($export_result.rc -ne 0) -or ((Get-Item -Path $secedit_ini_path).Length -eq 0)) {
+    Remove-Item -Path $secedit_ini_path
+    Fail-Json $result "Failed to export secedit.ini file to $($secedit_ini_path).`nRC: $($export_result.rc)`nSTDOUT: $($export_result.stdout)`nSTDERR: $($export_result.stderr)`nLOG: $($export_result.log)"
+}
+$secedit_ini = ConvertFrom-Ini -file_path $secedit_ini_path
+Remove-Item -Path $secedit_ini_path
+
+$will_change = $false
+if ($secedit_ini.$ini_right_key.ContainsKey($name)) {
+    $existing_users = Build-ExistingUserList -existing_users $secedit_ini.$ini_right_key.$name
+    $new_users = Build-NewUserList -action $action -users $users -existing_users $existing_users
+
+    # sort the user objects for later comparison and remove duplicates
+    $new_users = $new_users | Sort-Object -Unique
+    $existing_users = $existing_users | Sort-Object -Unique
```

Comment:
Shouldn't at least the uniqueness be handled inside the Build-XList function (if not also the sort)?